src/dmd/typesem.d

```diff
@@ -1231,11 +1231,6 @@ extern(C++) Type typeSemantic(Type t, Loc loc, Scope* sc)
             }
             if (tf.next.hasWild())
                 wildreturn = true;
-
-            if (tf.isreturn && !tf.isref && !tf.next.hasPointers())
```

Comment:
Why this change ?

Reply:
Per the 2nd paragraph in this PR's fist comment.

> Also, `return` was not emitting a similar error for the same reason as `scope`. I found that the compiler was actually removing the `return` attribute instead of emitting an error.

The compiler is currently silently removing the`return` attribute instead of emitting an error stating that the `return` attribute is not valid for functions. The compiler already emits an error for `scope` so this change makes `return` behave the same.